src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -294,5 +305,70 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
             }
             return donorName;
         }
+
+        /***************************************************************************************************
+        * @description sets DataImport__c record with matching url or failure information matching based on
+        * the BDI_Donations setDryRunDonationStatuses
+        * @param record: the DataImport__c record to check for matches
+        */
+        private void setDryRunRecordInfo(DataImport__c record) {
+            try {
+                String matchedUrl = '';
+                String matchedUrlLabel = '';
+                List<String> errors = new List<String>();
+
+                Boolean donationMatched = (record.DonationImportStatus__c == Label.bdiDryRunMatched ||
+                        record.DonationImportStatus__c == Label.bdiDryRunMatchedBest ||
+                        record.DonationImportStatus__c == Label.bdiDryRunMatchedId);
+                Boolean paymentMatched = (record.PaymentImportStatus__c == Label.bdiDryRunMatched ||
+                        record.PaymentImportStatus__c == Label.bdiDryRunMatchedBest ||
+                        record.PaymentImportStatus__c == Label.bdiDryRunMatchedId);
+                Boolean importFailure = record.FailureInformation__c != null;
+
+                if (paymentMatched) {
+                    //if there is a match, set url, preferably for the payment
+                    String importedId = record.PaymentImported__c != null ? record.PaymentImported__c : '';
```

Comment:
Could you move record.PaymentImport__c != null into the criteria for paymentMatched?